src/lib/comms/sol-lwm2m.c

```diff
@@ -4733,6 +5039,14 @@ sol_lwm2m_client_start(struct sol_lwm2m_client *client)
 
     SOL_NULL_CHECK(client, -EINVAL);
 
+    if (client->supports_access_control &&
+        client->need_to_setup_access_control) {
+        r = setup_access_control_object_instances(client);
+        SOL_INT_CHECK_GOTO(r, < 0, err_access_control);
```

Comment:
Be carefull, becuase if you jump to this goto, you will call  sol_lwm2m_resource_clear(&res[0]);, sol_lwm2m_resource_clear(&res[1]); and sol_lwm2m_resource_clear(&res[2]); Which are uninitialized. Thus can give as a headache